### PR TITLE
Fix BFT issues

### PR DESCRIPTION
**Pull request by ashamis**
This is work towards getting the partition tests to work correctly with BFT. We solve several issues here but not all of them so this PR is a checkpoint and more fixes will follow.

### CODE REVIEW

File: src/consensus/aft/impl/view_change_tracker.h [hunk 1]
```diff
@@ -258,6 +258,7 @@ namespace aft
         nv.view_change_messages.emplace(it.first, it.second);
       }
 
+      store->sign_view_change_confirmation(nv);
```
Comment (achamayou) [inline src/consensus/aft/impl/view_change_tracker.h:261]: Was this not signed at all before, or is this code being moved from somewhere else?

File: src/consensus/aft/raft.h [hunk 2]
```diff
@@ -436,14 +436,25 @@ namespace aft
...
       std::lock_guard<std::mutex> guard(state->lock);
-      return {get_term_internal(state->commit_idx), state->commit_idx};
+      if (consensus_type == ConsensusType::CFT)
+      {
+        return {get_term_internal(state->commit_idx), state->commit_idx};
+      }
+      else
+      {
+        auto progress_tracker = store->get_progress_tracker();
+        auto idx = progress_tracker->get_rollback_seqno();
```
Comment (achamayou) [inline src/consensus/aft/raft.h:446]: What's the rollback seqno, is the index of the last committed transaction?

Comment (ashamis) [reply src/consensus/aft/raft.h:446]: This is the latest prepared which is >= committed

File: src/consensus/aft/impl/view_change_tracker.h (hunk 1 above)
Comment (ashamis) [reply src/consensus/aft/impl/view_change_tracker.h:261]: yeah not signed before, so fun bug here :S

**Comment by ashamis** [comment]
Closing this PR as it has been broken down smaller PRs

**Pull request closed without merging**